src/positron-dts/positron.d.ts

```diff
@@ -1387,6 +1387,90 @@ declare module 'positron' {
 	}
 
 	namespace runtime {
+		/**
+		 * An object that observes an ongoing code execution invoked from the
+		 * `executeCode` API.
+		 */
+		export interface ExecutionObserver {
+			/**
+			 * An optional cancellation token that can be used to cancel the
+			 * execution.
+			 */
+			token?: vscode.CancellationToken;
+
+			/**
+			 * An optional callback to invoke when execution has started. This
+			 * may be different than the time `executeCode` was called, since
+			 * there may have been preceding statements in the queue, or we may
+			 * need to wait for the runtime to start or become ready.
+			 */
+			onStarted?: () => void;
+
+			/**
+			 * An optional callback to invoke when the execution emits text
+			 * output. This can be called zero or more times during execution of
+			 * the code.
+			 *
+			 * @param message The message emitted.
+			 */
+			onOutput?: (message: string) => void;
+
+			/**
+			 * An optional callback to invoke when the execution emits error
+			 * output. This just means "output sent to standard error", and does
+			 * not mean that the execution failed.
```

Comment:
```suggestion
			 * An optional callback to invoke when the execution emits error
			 * output. This just means "output sent to standard error", and does
			 * not mean that the execution failed. This can be called zero or more
			 * times during execution of the code.
```